casper/src/test/resources/ListOpsTest.rho

```diff
@@ -0,0 +1,120 @@
+//scalapackage coop.rchain.rholang.collection
+
+new
+  rl(`rho:registry:lookup`), ListOpsCh, RhoSpecCh,
+  stdlog(`rho:io:stdlog`),
+  greaterThan1,
+  setup,
+  test_all, test_all_is_lazy,
+  test_any, test_any_is_lazy
+in {
+
+  stdlog!("info", "Starting ListOps Test") |
+  rl!(`rho:id:zphjgsfy13h1k85isc8rtwtgt3t9zzt5pjd5ihykfmyapfc4wt3x5h`, *RhoSpecCh) |
+  for(@(_, RhoSpec) <- RhoSpecCh) {
+    @RhoSpec!("testSuite", *setup,
+      [
+        ("ListOps.all works", *test_all),
+        ("ListOps.all is lazy", *test_all_is_lazy),
+        ("ListOps.any works", *test_any),
+        ("ListOps.any is lazy", *test_any_is_lazy)
+      ])
+  } |
+  rl!(`rho:lang:listOps`, *ListOpsCh) |
+  for(@(_, ListOps) <- ListOpsCh) {
+
+    contract setup(returnCh) = {
+      returnCh!([])
+    } |
+
+    contract test_all(rhoSpec, _, ackCh) = {
+      new ch0, ch1, ch2, ch3, ch4, greaterThan1 in {
+        contract greaterThan1(@x, retCh) = { retCh!(x > 1)} |
+
+        @ListOps!("all", [], *greaterThan1, *ch0) |
+        @ListOps!("all", [1], *greaterThan1, *ch1) |
+        @ListOps!("all", [2], *greaterThan1, *ch2) |
+        @ListOps!("all", [3, 4], *greaterThan1, *ch3) |
+        @ListOps!("all", [1,2, 3, 4], *greaterThan1, *ch4) |
+
+        rhoSpec!("assertMany",
+          [
+            ((true, "== <-", *ch0), "all is successful for empty lists"),
+            ((false, "== <-", *ch1), ""),
+            ((true, "== <-", *ch2), ""),
+            ((true, "== <-", *ch3), ""),
+            ((false, "== <-", *ch4), "")
+          ], *ackCh)
+      }
+    } |
+
+    contract test_all_is_lazy(rhoSpec, _, ackCh) = {
```

Comment:
This is short_circuit not lazy

Reply:
Of course! I couldn't remember how it's called :)